polaris.shopify.com/src/styles/mixins.scss

```diff
@@ -28,3 +30,61 @@
     }
   }
 }
+
+@mixin responsive-props($componentName, $componentProp, $declarationProp) {
+  // stylelint-disable -- Polaris component custom properties
+  --pc-#{$componentName}-#{$componentProp}-xs: initial;
+  --pc-#{$componentName}-#{$componentProp}-sm: initial;
+  --pc-#{$componentName}-#{$componentProp}-md: initial;
+  --pc-#{$componentName}-#{$componentProp}-lg: initial;
+  --pc-#{$componentName}-#{$componentProp}-xl: initial;
+
+  #{$declarationProp}: var(--pc-#{$componentName}-#{$componentProp}-xs);
+
+  @media (min-width: $breakpointMobile) {
```

Comment:
breakpoint taxonomy and values are slightly different in the docs site. 
Let me know if these should be unified going forward, I can look at that work in a separate PR.

Reply:
I think this is good imo not critical to map to polaris-react